gateway/ln-gateway/src/lib.rs

```diff
@@ -275,44 +274,98 @@ impl Gateway {
 
         let scid_to_federation = self.scid_to_federation.clone();
-        let actors = self.actors.clone();
+        let clients = self.clients.clone();
         task_group
             .spawn(
                 "Subscribe to intercepted HTLCs in stream",
                 move |handle| async move {
                     // TODO: Need to recreate the lightning connection if it breaks while processing
                     // HTLCs
-                    while let Some(Ok(htlc)) = stream.next().await {
+                    while let Some(Ok(htlc_request)) = stream.next().await {
                         if handle.is_shutting_down() {
                             break;
                         }
 
                         let scid_to_feds = scid_to_federation.read().await;
-                        let federation_id = scid_to_feds.get(&htlc.short_channel_id);
-                        let outcome = {
-                            // Just forward the HTLC if we do not have a federation that
-                            // corresponds to the short channel id
-                            if let Some(federation_id) = federation_id {
-                                let actors = actors.read().await;
-                                let actor = actors.get(federation_id);
-                                // Just forward the HTLC if we do not have an actor that
-                                // corresponds to the federation id
-                                if let Some(actor) = actor {
-                                    actor.handle_intercepted_htlc(htlc).await
-                                } else {
-                                    InterceptHtlcResponse {
-                                        action: Some(Action::Forward(Forward {})),
-                                        incoming_chan_id: htlc.incoming_chan_id,
-                                        htlc_id: htlc.htlc_id,
+                        let federation_id = scid_to_feds.get(&htlc_request.short_channel_id);
+                        // Just forward the HTLC if we do not have a federation that
+                        // corresponds to the short channel id
+                        if let Some(federation_id) = federation_id {
+                            let clients = clients.read().await;
+                            let client = clients.get(federation_id);
+                            // Just forward the HTLC if we do not have a client that
+                            // corresponds to the federation id
+                            if let Some(client) = client {
+                                let htlc: Result<Htlc> = htlc_request.clone().try_into().map_err(|_| GatewayError::ClientNgError);
+                                if let Ok(htlc) = htlc {
+                                    let intercept_op = client.gateway_handle_intercepted_htlc(htlc).await;
+                                    // TODO: Refactor this into the state machine so we don't need to wait here
```

Comment:
This code is ugly and has lots of indentation. Ideally, I think the response to the lightning implementation should be inside the state machine, so that this loop does not need to wait for a response before handling the next HTLC. However, we can't do this without adding a separate `complete_htlc` RPC instead of using the bidirectional stream, since the bidirectional stream does not survive restarts. Will do this in a separate PR.